internal/response_assertions/api_versions_response_assertion.go

```diff
@@ -47,13 +47,19 @@ func (a *ApiVersionsResponseAssertion) ExpectApiKeyEntry(expectedApiKey int16, e
 	return a
 }
 
-func (a *ApiVersionsResponseAssertion) AssertSingleField(field field.Field) error {
+func (a *ApiVersionsResponseAssertion) AssertSingleField(field field.Field) *SingleFieldAssertionError {
```

Comment:
Is it necessary to change the return type here? The caller of `AssertSingleField` already has context of `field` so they can always construct `SingleFieldAssertionError` as needed after receiving the error from here. 

This change complicates the interface and implementation for every single response assertion's `AssertSingleField`, let's only do it if it's really worth it.